R/utilities-configuration-plan.R

```diff
@@ -117,12 +117,69 @@ getCurvePropertiesForTimeProfiles <- function(configurationPlanCurve,
 
   # Observed Data
   if (pathArray[2] %in% "ObservedData") {
-    observedData <- getObservedDataFromConfigurationPlan(pathArray[1], configurationPlan, logFolder)
+    observedResults <- getObservedDataFromConfigurationPlan(pathArray[1], configurationPlan, logFolder)
+    molWeight <- getMolWeightForObservedData(pathArray, configurationPlan, simulation)
+
+    time <- ospsuite::toUnit(
+      quantityOrDimension = "Time",
+      values = as.numeric(observedResults$data[, 1]),
+      targetUnit = axesProperties$x$unit,
+      sourceUnit = observedResults$metaData$time$unit
+    )
+    # Convert output values, if molWeight is NA but not required, then toUnit works without any issue
+    # if molWeight is NA and required, then toUnit crashes, error is caught
+    # and the error message indictes which observed data Id need molWeight
+    outputValues <- tryCatch({
+      ospsuite::toUnit(
+        quantityOrDimension = ospsuite::getDimensionForUnit(observedResults$metaData$output$unit),
+        values = observedResults$data[, 2],
+        targetUnit = axesProperties$y$unit,
+        sourceUnit = observedResults$metaData$output$unit,
+        molWeight = molWeight
+      )
+    },
+    error = function(e) {
+      NULL
+    }
+    )
+    if (isOfLength(outputValues, 0)) {
+      logErrorThenStop(
+        message = paste0(
+          "Molecular weight not found but required for observed data Id '", pathArray[1], "' in Time Profile plot."
+        ),
+        logFolderPath = logFolder
+      )
+    }
+
+    outputError <- NULL
+    if (!isOfLength(observedResults$metaData$error, 0)) {
+      # No unit means that error is geometric
+      outputError$ymin <- outputValues / observedResults$data[, 3]
+      outputError$ymax <- outputValues * observedResults$data[, 3]
+
+      if (!isIncluded(observedResults$metaData$error$unit, "")) {
+        outputError$ymin <- outputValues - ospsuite::toUnit(
+          ospsuite::getDimensionForUnit(observedResults$metaData$error$unit),

```

Comment:
same question as above

Reply:
Same as above, adding or removing a vector with NA elements leads to the NA elements in the same position in the result vector.

Reply:
and `ospsuite::toUnit` can also handle NaNs and will return NaN output value for NaN-input?

Reply:
I just checked, yes, that is the case